kasa/smartcam/detectionmodule.py

```diff
@@ -0,0 +1,51 @@
+"""SmartCamModule base class for all detections."""
+
+from __future__ import annotations
+
+import logging
+
+from kasa.feature import Feature
+from kasa.smart.smartmodule import allow_update_after
+from kasa.smartcam.smartcammodule import SmartCamModule
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DetectionModule(SmartCamModule):
+    """SmartCamModule base class for all detections."""
+
+    DETECTION_FEATURE_ID: str = ""
+    DETECTION_FEATURE_NAME: str = ""
+    QUERY_SETTER_NAME: str = ""
+    QUERY_SET_SECTION_NAME: str = ""
```

Comment:
I would add a quick comment here that these are to be filled by inheriting classes.